Добавлено отображение никнейма в сообщение о взаимном лайке

NRenat opened the pull request:
- Также добавлена возможность перейти к переписке с пользователями, у которых не был установлен никнейм в телегаме.

NRenat committed:
Добавлено отображение никнейма в сообщение о взаимном лайке
- Также добавлена возможность перейти к переписке с пользователями, у которых не был установлен никнейм в телегаме.

NRenat commented:
Тестил отображение никнейма на новом тг акке, где не настраивал никнейм. Бот вел к нему через @None. 
ps "@" перенес из шаблона в функцию, чтобы для пользователей без никнейма "@" так же подсвечивалась, хотя там она не функциональна.

TomatoInOil commented on src/bot/conversations/match_requests/utils.py:
```diff
@@ -13,6 +14,22 @@ async def send_match_notifications(update, context, sender_id, receiver_id):
+            f'<a href="tg://user?id={receiver_id}">@{user.name}</a>'
+        )
+
+    if like_sender_username:
+        like_sender_username = f"@{like_sender_username}"
+    else:
+        user = await api_service.get_user_profile_by_telegram_id(telegram_id=sender_id)
+        like_sender_username = f'<a href="tg://user?id={sender_id}">@{user.name}</a>'
```
Предлагаю эту логику поместить в функцию `_get_tg_username`, немного её переименовав, это избавит от дублирования и улучшит читаемость кода 
```Python
async def _get_username_or_profile_link(
    context: ContextTypes.DEFAULT_TYPE, telegram_id: int
) -> str:
    chat = await context.bot.get_chat(chat_id=telegram_id)
    if chat.username:
        return f"@{chat.username}"
    if chat.first_name:  # Можно взять имя из Telegram, если оно есть
        return f'<a href="tg://user?id={telegram_id}">@{chat.first_name}</a>'
    profile = await api_service.get_user_profile_by_telegram_id(telegram_id=telegram_id)
    return f'<a href="tg://user?id={telegram_id}">@{profile.name}</a>'
```

TomatoInOil replied:
https://github.com/Studio-Yandex-Practicum/coliving_bot/blob/0883e2ddec3f8429df8a40236115320a74c21074/src/bot/conversations/match_requests/utils.py#L26

NRenat committed:
review
- Перенесена логика получения тега пользователя в новую _get_username_or_profile_link.
